ARKitExample/ViewController.swift

```diff
@@ -0,0 +1,218 @@
+//
+//  ViewController.swift
+//  ARKitExample
+//
+//  Copyright © 2016-2017 Twilio, Inc. All rights reserved.
+//
+
+import UIKit
+import SceneKit
+import ARKit
+import TwilioVideo
+
+class ViewController: UIViewController {
+
+    @IBOutlet var sceneView: ARSCNView!
+
+    override var prefersStatusBarHidden: Bool {
+        return true
+    }
+
+    var accessToken = "TWILIO_ACCESS_TOKEN"
+    var room: TVIRoom?
+    weak var consumer: TVIVideoCaptureConsumer?
+    var frame: TVIVideoFrame?
+    var displayLink: CADisplayLink?
+
+    var videoTrack: TVILocalVideoTrack?
+    var audioTrack: TVILocalAudioTrack?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Configure the ARSCNView which will be used to display the AR content.
+        sceneView.delegate = self
+
+        // Since we will also capture from the view we will limit ourselves to 30 fps.
+        sceneView.preferredFramesPerSecond = 30
+        // Since we are in a streaming environment, we will render at a relatively low resolution.
+        sceneView.contentScaleFactor = 1
+
+        // Show feature points, and statistics such as fps and timing information
+        sceneView.showsStatistics = true
+        sceneView.debugOptions =
+            [ARSCNDebugOptions.showFeaturePoints, ARSCNDebugOptions.showWorldOrigin]
+
+        // Create a new scene, and bind it to the view.
+        let scene = SCNScene(named: "art.scnassets/ship.scn")!
+        sceneView.scene = scene
+
+        self.videoTrack = TVILocalVideoTrack.init(capturer: self)
+        self.audioTrack = TVILocalAudioTrack.init()
+
+        let options = TVIConnectOptions.init(token: accessToken, block: {(builder: TVIConnectOptionsBuilder) -> Void in
+            if let videoTrack = self.videoTrack {
+                builder.videoTracks = [videoTrack]
+            }
+            if let audioTrack = self.audioTrack {
+                builder.audioTracks = [audioTrack]
+            }
+            builder.roomName = "arkit"
+        })
+
+        self.room = TwilioVideo.connect(with: options, delegate: self)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Release any cached data, images, etc that aren't in use.
+    }
+
+    @objc func displayLinkDidFire() {
```

Comment:
I know we said we were gonna deal with the Swift idioms later... But this is a little more idiomatic if you'd like to drop it in:

```
    @objc func displayLinkDidFire() {
        // Our capturer polls the ARSCNView's snapshot for processed AR video content, and then copies the result into a CVPixelBuffer.
        // This process is not ideal, but it is the most straightforward way to capture the output of SceneKit.
        let myImage = self.sceneView.snapshot

        guard let imageRef = myImage().cgImage else {
            return
        }

        // As a TVIVideoCapturer, we must deliver CVPixelBuffers and not CGImages to the consumer.
        if let pixelBuffer = self.copyPixelbufferFromCGImageProvider(image: imageRef) {
            self.frame = TVIVideoFrame(timestamp: Int64((displayLink?.timestamp)! * 1000000),
                                       buffer: pixelBuffer,
                                       orientation: TVIVideoOrientation.up)
            self.consumer?.consumeCapturedFrame(self.frame!)
        }
    }
```